File: backend/models/line_item.py
# Line Item Pydantic Models

from pydantic import BaseModel, Field
from typing import Optional, List
from enum import Enum


class CodeType(str, Enum):
    CPT = "CPT"
    HCPCS = "HCPCS"
    REV = "REV"
    ICD = "ICD"


class Category(str, Enum):
    FACILITY = "facility"
    PHYSICIAN = "physician"
    LAB = "lab"
    IMAGING = "imaging"
    PROCEDURE = "procedure"
    MEDICATION = "medication"
    SUPPLY = "supply"
    OTHER = "other"


class RiskLevel(str, Enum):
    NORMAL = "normal"
    NEEDS_REVIEW = "needs_review"
    HIGH_RISK = "high_risk"


class Severity(str, Enum):
    INFO = "info"
    WARNING = "warning"
    CRITICAL = "critical"


class LineItemFlag(BaseModel):
    type: str
    message: str
    severity: Severity
    suggested_action: str


class LineItemResponse(BaseModel):
    id: str = Field(alias="_id")
    bill_id: str
    service_date: Optional[str] = None
    description: str
    code: Optional[str] = None
    code_type: Optional[CodeType] = None
    quantity: int = 1
    billed_amount: float
    allowed_amount: Optional[float] = None
    insurance_paid: Optional[float] = None
    patient_responsibility: Optional[float] = None
    adjustment_reason: Optional[str] = None
    category: Optional[Category] = None
    confidence: float = 0.0
    risk_level: RiskLevel = RiskLevel.NORMAL
    flags: List[LineItemFlag] = Field(default_factory=list)

    class Config:
        populate_by_name = True


class FieldCorrection(BaseModel):
    line_item_id: Optional[str] = None
    field: str
    corrected_value: str

File: src/app/types/line-item.ts
// LineItem types — LineItem, LineItemFlag, CodeType, Category, RiskLevel

export type CodeType = "CPT" | "HCPCS" | "REV" | "ICD" | null;
export type Category = "facility" | "physician" | "lab" | "imaging" | "procedure" | "medication" | "supply" | "other";
export type RiskLevel = "normal" | "needs_review" | "high_risk";
export type FlagType = "duplicate" | "overpriced" | "date_mismatch" | "denied_billed" | "quantity_anomaly" | "unbundled";
export type Severity = "info" | "warning" | "critical";

export interface LineItemFlag {
  type: FlagType;
  message: string;
  severity: Severity;
  suggested_action: string;
}

export interface CallResolution {
  status: "resolved" | "adjusted" | "denied";
  previous_amount: number;
  new_amount: number;
  note: string;
  call_id: string;
}

export interface LineItem {
  _id: string;
  bill_id: string;
  service_date: string | null;
  description: string;
  code: string | null;
  code_type: CodeType;
  quantity: number;
  billed_amount: number;
  allowed_amount: number | null;
  insurance_paid: number | null;
  patient_responsibility: number | null;
  adjustment_reason: string | null;
  category: Category;
  confidence: number;
  risk_level: RiskLevel;
  flags: LineItemFlag[];
  call_resolution?: CallResolution;
}
